src/usage_metrics/scripts/save_zenodo_metrics.py

```diff
@@ -58,55 +44,41 @@ def check_empty_string(cls, doi: str):  # noqa: N805
 
 
 def get_zenodo_logs() -> pd.DataFrame():
-    """Get all metrics for all versions of records in the Catalyst Cooperative Zenodo community."""
+    """Download JSONs of metrics for all versions in the Catalyst Cooperative Zenodo community."""
+    bucket_name = "pudl-usage-metrics-archives.catalyst.coop"
+    client = storage.Client()
+    bucket = client.get_bucket(bucket_name)
+
     community_url = "https://zenodo.org/api/communities/14454015-63f1-4f05-80fd-1a9b07593c9e/records"
     # First, get metadata on all the datasets in the Catalyst Cooperative community
-    community_records = requests.get(community_url, timeout=100)
+    community_records = make_request(url=community_url)
     dataset_records = community_records.json()["hits"]["hits"]
     dataset_records = [CommunityMetadata(**record) for record in dataset_records]
-    stats_dfs = []
 
     for record in dataset_records:
         logger.info(f"Getting usage metrics for {record.title}")
         # For each dataset in the community, get all archived versions and their
         # corresponding metrics.
         versions_url = f"https://zenodo.org/api/records/{record.recid}/versions"
-        record_versions = requests.get(versions_url, timeout=100)
-        version_records = record_versions.json()["hits"]["hits"]
-        version_records = [
-            CommunityMetadata(**version_record) for version_record in version_records
-        ]
-        version_df = pd.DataFrame(
-            [
-                dict(
-                    version_record.stats.dict(),
-                    doi=version_record.doi,
-                    title=version_record.title,
-                )
-                for version_record in version_records
-            ]
-        )
-        if not version_df.empty:
-            stats_dfs.append(version_df)
-    return pd.concat(stats_dfs)
-
-
-def upload_to_bucket(data: pd.DataFrame) -> None:
-    """Upload a gcp object."""
-    bucket_name = "pudl-usage-metrics-archives.catalyst.coop"
-    client = storage.Client()
-    bucket = client.get_bucket(bucket_name)
+        record_versions = make_request(url=versions_url).json()
+        versions_metadata = json.dumps(record_versions)
+        blob_name = f"zenodo/{date.today().strftime('%Y-%m-%d')}-{record.recid}.json"
+        upload_to_bucket(bucket=bucket, blob_name=blob_name, data=versions_metadata)
```

Comment:
Nit: The `get_` and `save_` verbs imply that one function merely downloads and the other persists the data - which was the old behavior. Do you think it's worth moving the "upload to bucket" logic out of the `get` function and into `save`?

Reply:
I just scrapped one of the functions to simplify.